static/src/js/components/Sidebar/Sidebar.jsx

```diff
@@ -11,16 +13,34 @@ const Sidebar = ({
   visible,
   headerChildren
 }) => {
+  const setSidebarWidth = useEdscStore((state) => state.ui.panels.setSidebarWidth)
   const className = classNames({
     sidebar: true,
     'sidebar--hidden': !visible
   })
 
+  const observedElementRef = useRef(null)
+
+  useEffect(() => {
+    // Use ResizeObserver to detect changes in the size of the observedElementRef (the sidebar)
+    const observer = new ResizeObserver((entries) => {
+      entries.forEach((entry) => {
+        setSidebarWidth(entry.contentRect.width)
```

Comment:
Just looked at differences between `offsetWidth` and `getBoundingClientRect().width` because I was curious:

> `offsetWidth` returns the layout width before any CSS transforms are applied, while getBoundingClientRect().width returns the width after transforms are applied, reflecting the element's visual size on the screen.

We don't have any transforms happening here but it might be more future-proofed to go with getBoundingClientRect().width